packages/hooks/src/useAppHandler/index.ts

```diff
@@ -0,0 +1,437 @@
+import {
+  createParser,
+  ParsedEvent,
+  ReconnectInterval,
+} from 'eventsource-parser';
+
+import { encreCache } from '../cache';
+import {
+  Context,
+  EncreAppConfig,
+  EncreAppHander,
+  EncreAppInputHandler,
+  EncreAppOutputHandler,
+  EncreData,
+  Message,
+  ProcessStreamEventFilter,
+} from '../types';
+import { coerceEncreData } from '../utils';
+
+const runGraphFetch = async (
```

Comment:
Just to make sure I understand this, is the purpose of this reads what's in the graph and populate important information in Record<> form?

Reply:
and if the callbacks are unknown, what are the purposes of these callbacks?

Reply:
Can you reference the code chunk? I need know what callbacks you are asking here.

Reply:
For example, this:

nodeCallbacks: {
    [key in string]: {
      onNodeStart?: (
        nodeId: string,
        nodeTitle: string,
        inputs: Record<string, EncreData>,
      ) => unknown;
      onNodeFinish?: (
        nodeId: string,
        nodeTitle: string,
        outputs: Record<string, EncreData>,
      ) => unknown;
    };
  },
  graphCallbacks: {
    onDone?: (
      graphOutput: Record<string, Record<string, EncreData>>,
    ) => unknown;
    onError?: (error: string) => unknown;
  },
  
  What are these use for?

Reply:
These callbacks are coming from users. Basically, when user called `onOutput` function, they can write their own `onRead`, `onDone`, `onError` callbacks. Those callbacks will be formatted into the corresponding `nodeCallbacks`, `graphCallbacks` based on the hooks logic. So we won't know the output of those callbacks are. However, we will coerce those outputs so that we can ensure the data is formatted correctly to those output Components.

Reply:
The Hooks documentation may have more detailed description for those callbacks.